93-appendixC.Rmd

```diff
@@ -1,69 +1,351 @@
-ggplot(data = flights, mapping = aes(x = carrier)) +
-  geom_bar() +
-  scale_x_discrete(limits = names(sorted_flights))
+ggplot(revenue_by_type, aes(x = type, y = total_revenue)) +
+  geom_col() + 
+  labs(x = "Movie genre", y = "Total boxoffice revenue (in millions of $)")
 ```
 
-```{r include=FALSE}
-#ggplot(data = carrier_counts, mapping = aes(x = carrier, y = n)) +
-#  geom_bar(stat = "identity") + 
-#  scale_x_discrete(limits = carrier)
+Or if you want to reorder `type` in ascencding order of `total_revenue`, we use `reorder()`
+
+```{r}
+revenue_by_type <- revenue_by_type %>% 
+  mutate(type = reorder(type, total_revenue))
+
+ggplot(revenue_by_type, aes(x = type, y = total_revenue)) +
+  geom_col() + 
+  labs(
+    x = "Movie genre", y = "Total boxoffice revenue (in millions of $)"
+  )
 ```
 
-The last addition here specifies the values of the horizontal `x` axis on a discrete scale to correspond to those given by the entries of `sorted_flights`.  
+Or if you want to reorder `type` in descencding order of `total_revenue`, just put
+a `-` sign in front of `-total_revenue` in `reorder()`:
 
-## Interactive graphics
+```{r}
+revenue_by_type <- revenue_by_type %>% 
+  mutate(type = reorder(type, -total_revenue))
+
+ggplot(revenue_by_type, aes(x = type, y = total_revenue)) +
+  geom_col() + 
+  labs(
+    x = "Movie genre", y = "Total boxoffice revenue (in millions of $)"
+  )
+```
+
+For more advanced categorical variable (i.e. factor) manipulations, check out
+the [`forcats` package](https://forcats.tidyverse.org/){target="_blank"}. Note: `forcats` is an anagram of `factors`
+
+
+![](https://github.com/tidyverse/forcats/raw/master/man/figures/logo.png){width=20%} 
+
+
+
+### Showing money on an axis
+
+```{r fig.height=3, fig.width=5}
+movies <- movies %>% 
+  mutate(revenue = millions * 10^6)
+
+ggplot(data = movies, aes(x = rating, y = revenue)) + 
+  geom_boxplot() + 
+  labs(x = "rating", y = "Revenue in $", title = "Profits for different movie ratings")
+```
+
+Google "ggplot2 axis scale dollars" and click on the [first link](http://www.sthda.com/english/wiki/ggplot2-axis-scales-and-transformations) and search for the word "dollars". You'll find:
+
+```{r fig.height=3, fig.width=5}
+# Don't forget to load the scales package first!
+library(scales)
+
+ggplot(data = movies, aes(x = rating, y = revenue)) + 
+  geom_boxplot() + 
+  labs(x = "rating", y = "Profit in millions of $", title = "Profits for different movie ratings") +
+  scale_y_continuous(labels = dollar)
+```
+
+
+
+### Changing values inside cells
+
+`dplyr::rename()` renames column/variable names. To "rename" values inside cells of a particular column, you need to `mutate()` the column using one of the three functins below. There might be other ones too, but these are the three I've seen the most. In these examples, we'll change values in the variable `type`.
+
+1. `ifelse()`
+1. `recode()`
+1. `case_when()`
+
+#### `ifelse()` {-}
```

Comment:
Probably best to switch this to use `dplyr::if_else()` instead here

Reply:
sounds good